### PR TITLE
Serge

### PR DESCRIPTION
**Pull request by Sergej-Karyuhin**

### CODE REVIEW

File: src/app/FastSearchMovie/FastSearchMovie.component.html [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<div *ngFor="let item of searchItems" class="fast-search__item">
+  <img [src]="item.url" alt="" class="img">
+  <div class="item-wrapper">
+    <p class="item__title">{{item.name}}</p>
+    <p class="item__description">{{item.description}}</p>
+  </div>
```
Comment (Vladissqq) [inline src/app/FastSearchMovie/FastSearchMovie.component.html:6]: Can u take out it in separate component?

File: src/app/FastSearchMovie/FastSearchMovie.component.html [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<div *ngFor="let item of searchItems" class="fast-search__item">
```
Comment (Vladissqq) [inline src/app/FastSearchMovie/FastSearchMovie.component.html:1]: ngFor contain array of items. This class name does not fit in this case

File: src/app/FastSearchMovie/FastSearchMovie.component.scss [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    background: rgba(248, 248, 248, 0.32);
+  }
+
+  .img {
+    display: flex;
+    align-items: center;
+    width: 131px;
+    height: 74px;
+    border-radius: 5px;
+    background-color: yellow;
```
Comment (Vladissqq) [inline src/app/FastSearchMovie/FastSearchMovie.component.scss:20]: Use variables for colors

File: src/app/FilmCategories/FilmCategories.component.html [hunk 4]
```diff
@@ -0,0 +1,6 @@
+<ul>
+  <li *ngFor="let category of categories">
+    <span class="title">{{category.name}}</span>
+    <MoviesCategories></MoviesCategories>
```
Comment (Vladissqq) [inline src/app/FilmCategories/FilmCategories.component.html:4]: I would create a wrapper for this block

File: src/app/MoviesCategories/MoviesCategories.component.html [hunk 5]
```diff
@@ -0,0 +1,31 @@
+<ul class="category-list">
+    <li *ngFor="let movie of movies"
```
Comment (Vladissqq) [inline src/app/MoviesCategories/MoviesCategories.component.html:2]: Again. I would move  code below in separate component

File: src/app/MoviesCategories/MoviesCategories.component.html [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+        (click)="onSelect(movie)"
+        [class.selected]="movie === selectedMovie"
+        [ngStyle]="{background: 'url(' + movie.url + ')' + ' center/100%'}"
```
Comment (Vladissqq) [inline src/app/MoviesCategories/MoviesCategories.component.html:5]: You can move style obj in component

File: src/app/MoviesCategories/MoviesCategories.component.html [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+        class="category-list__item" >
+
+        <section class="item__header">
+          <p class="title title_selected">{{movie.name}}</p>
```
Comment (Vladissqq) [inline src/app/MoviesCategories/MoviesCategories.component.html:9]: Is this item title, or its title for Main page or for category ?

File: src/app/MoviesCategories/MoviesCategories.component.html [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+          <div [class.hidden]="!(movie === selectedMovie)" class="header__imdb">
+            <p class="imdb__title">IMDb 8,5</p>
+            <img src="https://cdn.iconscout.com/icon/free/png-256/star-bookmark-favorite-shape-rank-like-32386.png" alt="star" width="16px" height="16px">
+          </div>
+        </section>
+        <section [class.hidden]="!(movie === selectedMovie)" class="item__categories" >
+          <div class="categories__item"><p class="text">Action</p></div>
+          <div class="categories__item"><p class="text">Adventure</p></div>
+          <div class="categories__item"><p class="text">Drama</p></div>
```
Comment (Vladissqq) [inline src/app/MoviesCategories/MoviesCategories.component.html:18]: Dont repeat yourself😂

File: src/app/MoviesCategories/MoviesCategories.component.html [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+        (click)="onSelect(movie)"
+        [class.selected]="movie === selectedMovie"
+        [ngStyle]="{background: 'url(' + movie.url + ')' + ' center/100%'}"
+        class="category-list__item" >
+
+        <section class="item__header">
+          <p class="title title_selected">{{movie.name}}</p>
+          <div [class.hidden]="!(movie === selectedMovie)" class="header__imdb">
+            <p class="imdb__title">IMDb 8,5</p>
+            <img src="https://cdn.iconscout.com/icon/free/png-256/star-bookmark-favorite-shape-rank-like-32386.png" alt="star" width="16px" height="16px">
```
Comment (Vladissqq) [inline src/app/MoviesCategories/MoviesCategories.component.html:12]: Use constants for links

File: src/app/MoviesCategories/MoviesCategories.component.ts [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+import { WatchlistService } from '../services/watchlist.service';
+
+@Component({
+  selector: 'MoviesCategories',
+  templateUrl: './MoviesCategories.component.html',
+  styleUrls: ['./MoviesCategories.component.scss']
+})
+export class MoviesOfTheCategoriesComponent implements OnInit {
+  movies = MOVIES;
+  selectedMovie: Movie;
```
Comment (Vladissqq) [inline src/app/MoviesCategories/MoviesCategories.component.ts:15]: is this public, or private variables?

File: src/app/services/watchlist.service.ts [hunk 11]
```diff
@@ -0,0 +1,18 @@
+import { Injectable } from '@angular/core';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class WatchlistService {
+  constructor() { }
+
+  setMovie(movie) {
+    if (!localStorage.getItem(movie)) {
```
Comment (Vladissqq) [inline src/app/services/watchlist.service.ts:10]: i would keep films in array on LS

File: src/app/FastSearchMovie/FastSearchMovie.component.scss (hunk 3 above)
Comment (marinbonibon) [reply src/app/FastSearchMovie/FastSearchMovie.component.scss:20]: Or hex color code if you use it only in one place

**Pull request closed without merging**